command.go

```diff
@@ -943,9 +949,7 @@ func (c *Command) ExecuteC() (cmd *Command, err error) {
 
 	// We have to pass global context to children command
 	// if context is present on the parent command.
-	if cmd.ctx == nil {
-		cmd.ctx = c.ctx
-	}
+	cmd.ctx = c.ctx
```

Comment:
Why was the `if` removed here?

I can imagine, even with the addition of the new `SetContext` API, there still being the need for this if catch. The way I read this code now is that _on every execution,_ a passed in command will _always_ have it's context set.

This could even be a breaking change if during command execution, the users rely on the `cmd.ctx` not trickling down to subsequent child commands. This change makes the assumption that _all_ sub-command executions should have the same context as the parent.

Reply:
Please see  #1109 for the motivation behind this change.